processing/src/main/java/org/apache/druid/query/scan/ScanResultValueFramesIterable.java

```diff
@@ -200,26 +229,33 @@ public FrameSignaturePair next()
       // start all the processing
       populateCursor();
       boolean firstRowWritten = false;
-      // While calling populateCursor() repeatedly, currentRowSignature might change. Therefore we store the signature
+      // While calling populateCursor() repeatedly, currentRowSignature might change. Therefore, we store the signature
```

Comment:
....what if the signature changes - is that a problem? shouldn't that be an Exception?

Reply:
if there are two cursors, CursorA with RowSignatureA and CursorB with RowSignatureB and the cursor is at the last row of CursorA, populate call will return `false`, i.e. the two cursors cannot be batched together, and set `currentRowSignature` to the RowSignatureB (i.e. prepare the variables for the next write). We still want to return the old frame with the old signature therefore we need to preserve the signature with which we have written the frame. 
Per your previous suggestion, `frameWriterFactory.signature()` would be sufficient and cleaner, and I will use that instead.